Yona/Yona/BaseViewController.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  BaseViewController.swift
+//  Yona
+//
+//  Created by Chandan on 19/05/16.
+//  Copyright © 2016 Yona. All rights reserved.
+//
+
+import UIKit
+
+class BaseViewController: UIViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(BaseViewController.foreground), name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
```

Comment:
Then this should not be managed by the view controller it self.
Also, this will hold the view controller in the memory; and there is no place I see it is being removed.